src/js/playlist.js

```diff
@@ -6,12 +6,11 @@ class Playlist {
 		const currentId = opts.player.videoData ? opts.player.videoData.id : opts.player.opts.id;
 		this.currentIndex = currentId ? opts.queue.indexOf(currentId.toString()) : -1;
 
-		this.opts.player.containerEl.addEventListener('ended', this.next.bind(this), true);
-
 		this.cache = {};
 
-		if (this.opts.autoplay && this.currentIndex === -1) {
-			this.next();
+		if (this.opts.autoplay) {
+			this.opts.player.containerEl.addEventListener('ended', this.next.bind(this), true);
+			this.currentIndex === -1 && this.next();
```

Comment:
That search return every multi-condition ever and it doesn't so I'll blame the search